optaplanner-website-root/content/blog/2023/04/05/BoostVRPListSelectorScalability.adoc

```diff
@@ -0,0 +1,102 @@
+You can find out more about optaplanner-benchmarks https://www.optaplanner.org/docs/optaplanner/latest/benchmarking-and-tweaking/benchmarking-and-tweaking.html[here].
+
+== Execution time
+
+How fast each algorithm improves the solution score?
+
+Graph below represents the speed of score calculation for different selector options for a certain amount of samples.
+The x-axis represents the size of datasets from 50 customers with 10 vehicles to 1000 customers with 20 vehicles while the y-axis represents the ratio of score calculation speed = best score/time spent.
+
+The results indicate that the LANS consistently outperforms the LA algorithm.
+In particular, you can see that LANS with planning List Variable parabolic distribution (LV Nearby 20) was the fastest on average, while the LA was the slowest.
+
+image::scoreCalculationSpeedPerSecondNearby.png[How NearByMoveSelector improves score calculation speed]
+
+[#table1]
+.Score calculation count per second comparing LANS vs LA
+|===
+| ^.^|Belgium-n50-k10
+(2,890.00 samples)|Belgium-n100-k10 (10,790.00 samples)|Belgium-n500-k20
+(258,980.00 samples)|Belgium-n1000-k20 (1,017,980.00 samples)
+|Late Acceptance                      >|35,393.00/s >|37,334.00/s >|31,667.00/s >|30,282.00/s
+|LV Nearby Parabolic Distribution 10  >|40,428.00/s >|45,059.00/s >|32,708.00/s >|32,036.00/s
+|LV Nearby Parabolic Distribution 20  >|38,567.00/s >|40,552.00/s >|34,418.00/s >|37,188.00/s
+|LV Nearby Parabolic Distribution 40  >|37,551.00/s >|39,833.00/s >|33,558.00/s >|35,125.00/s
+|LV Nearby Parabolic Distribution 80  >|37,428.00/s >|35,625.00/s >|30,523.00/s >|36,060.00/s
+|LV Nearby Parabolic Distribution 160 >|41,197.00/s >|38,086.00/s >|33,724.00/s >|34,451.00/s
+|===
+
+== Solution quality
+
+Despite the fact that NS improves the speed of calculation, it could potentially decrease the quality of the best solution.
+
+image::bestScoreNB.png[How NS affects scores]
+
+The image above represents the overall best score archived (y-axis) with different move selectors and datasets for the same amount of time (5 minutes).
+The less is better.
+
+You may see how much the NS improves the score calculation while the dataset increases.
+For the biggest dataset any LANS variation gain twice better score then LA.
+
+From the other side look at the LV Nearby 10 bars.
+Despite being the fastest solution of all with the small dataset, it somehow dramatically loose the score gain for the case of 500 customers and 20 vehicles.
+
+The graph below shows how in this case best score (y-axis) changes in time during the calculation . The less is better.
+
+image::belgium-n500-k20BestScoreStatisticLevel.png[How local optimum affects the score]
+
+You can clearly see that at some point due to the small distribution size it fails to reach better solution and not able to get out of the local optimum providing the same result as LA.
+
+== Scalability
+
+image::bestScoreScalabilitySummaryLevel1.png[Scalability of NS solver]
```

Comment:
I like the graph. Especially the description of axes in it. :-)

Reply:
I cannot recognize the irony here)) I removed the description of axes 38291a34

Reply:
No irony intended; when I was writing the comment, this graph already had the correct axes description.